src/osx/cocoa/settings.mm

```diff
@@ -0,0 +1,243 @@
+    NSColor* sysColor = nil;
+    switch( index )
+    {
+    case wxSYS_COLOUR_SCROLLBAR:
+        sysColor = [NSColor scrollBarColor]; // color of slot
+        break;
+    case wxSYS_COLOUR_DESKTOP: // No idea how to get desktop background
+        // fall through, window background is reasonable
+    case wxSYS_COLOUR_ACTIVECAPTION: // No idea how to get this
+        // fall through, window background is reasonable
+    case wxSYS_COLOUR_INACTIVECAPTION: // No idea how to get this
+        // fall through, window background is reasonable
+    case wxSYS_COLOUR_MENU:
+    case wxSYS_COLOUR_MENUBAR:
+    case wxSYS_COLOUR_WINDOW:
+    case wxSYS_COLOUR_WINDOWFRAME:
+    case wxSYS_COLOUR_ACTIVEBORDER:
+    case wxSYS_COLOUR_INACTIVEBORDER:
+    case wxSYS_COLOUR_GRADIENTACTIVECAPTION:
+    case wxSYS_COLOUR_GRADIENTINACTIVECAPTION:
+        sysColor = [NSColor windowFrameColor];
+        break;
+    case wxSYS_COLOUR_BTNFACE:
+        sysColor = [NSColor controlColor];
+        break;
+    case wxSYS_COLOUR_LISTBOX:
+        sysColor = [NSColor controlBackgroundColor];
+        break;
+    case wxSYS_COLOUR_BTNSHADOW:
+        sysColor = [NSColor controlShadowColor];
+        break;
+    case wxSYS_COLOUR_BTNTEXT:
+    case wxSYS_COLOUR_MENUTEXT:
+    case wxSYS_COLOUR_WINDOWTEXT:
+    case wxSYS_COLOUR_CAPTIONTEXT:
+    case wxSYS_COLOUR_INFOTEXT:
+    case wxSYS_COLOUR_INACTIVECAPTIONTEXT:
+    case wxSYS_COLOUR_LISTBOXTEXT:
+        sysColor = [NSColor controlTextColor];
+        break;
+    case wxSYS_COLOUR_HIGHLIGHT:
+        sysColor = [NSColor selectedControlColor];
+        break;
+    case wxSYS_COLOUR_BTNHIGHLIGHT:
+        sysColor = [NSColor controlHighlightColor];
+        break;
+    case wxSYS_COLOUR_GRAYTEXT:
+        sysColor = [NSColor disabledControlTextColor];
+        break;
+    case wxSYS_COLOUR_3DDKSHADOW:
+        sysColor = [NSColor controlShadowColor];
+        break;
+    case wxSYS_COLOUR_3DLIGHT:
+        sysColor = [NSColor controlHighlightColor];
+        break;
+    case wxSYS_COLOUR_HIGHLIGHTTEXT:
+    case wxSYS_COLOUR_LISTBOXHIGHLIGHTTEXT:
+        sysColor = [NSColor alternateSelectedControlTextColor];
+        break;
+    case wxSYS_COLOUR_INFOBK:
+        // tooltip (bogus)
+        sysColor = [NSColor windowFrameColor];
+        break;
+    case wxSYS_COLOUR_APPWORKSPACE:
+        // MDI window color (bogus)
+        sysColor = [NSColor windowBackgroundColor];
+        break;
+    case wxSYS_COLOUR_HOTLIGHT:
+        // OSX doesn't change color on mouse hover
+        sysColor = [NSColor controlTextColor];
+        break;
+    case wxSYS_COLOUR_MENUHILIGHT:
+        sysColor = [NSColor selectedMenuItemColor];
+        break;
+    case wxSYS_COLOUR_MAX:
+    default:
+        if(index>=wxSYS_COLOUR_MAX)
+        {
+            wxFAIL_MSG(wxT("Invalid system colour index"));
+            return wxColour();
+        }
+    }
+    
+    if ( sysColor )
+    {
+        CGColorRef cgCol = sysColor.CGColor;
```

Comment:
The CGColor property unfortunately is available since 10.8 while wx currently supports 10.7+.
If wxColour's ctor with NSColor isn't being used here (as it was in wxCocoa) because it is less accurate than using CGColor, then perhaps this should be made conditional?
I actually tried the ctor with NSColor just now (e.g. ``return wxColour([NSColor windowFrameColor]);``) but that causes an exception which I don't currently understand.